scripts/python/test.sh

```diff
@@ -28,3 +28,4 @@ else
   pytest "$@"
 fi
 
+sh "$ROOT_DIR"/scripts/python/test_residue_cleanup.sh
```

Comment:
Run it inside a trap instead so it's called in case of failure?
```

cleanup_handler() {
  "$ROOT_DIR"/scripts/python/test_residue_cleanup.sh
}

trap cleanup_handler ERR INT QUIT TERM HUP
```